### ISSUE TITLE
Unwanted tab-completion for sub file

### ISSUE THREAD

**Issue by luigino** [2ion/mpv-bash-completion#19]
* mpv version: 0.27.2
 * mpv-bash-completion version (or link to package): 3.3.16
 * Operating system: openSUSE Leap 15.0

---------

If both foo.mkv and foo.srt are present, tab-completion cycles between the two of them.
Would it be possible to have the .srt only be considered after the --sub-file switch?
Using bash 4.4.19.

**Comment by 2ion** [comment]
The trouble with filtering arguments to options that take file or directory paths by file name extension is that almost any attempt to create reasonable filter lists for all of them is a futile task.

This starts with deciding what kind of extensions sub-file should look at: srt, ass, lrc, smi, usf, and there are probably more since ffmpeg can read a lot of things – just limiting results to the most common extensions is bound to deliver incomplete results in some cases.

As beside sub-file, there are currently 71 options that take file paths excluding the main media file argument in mpv 0.28, I would prefer to not implement filtering at all since this would be a patchwork effort and, as explained, full of unresolved edge cases.

Especially since for subtitle files, mpv should pick up files with the same stem automatically (like  considering x.srt for x.mkv) if configured correctly, so please consider adding the sub-auto option with appropriate arguments to your mpv configuration file; this should allow you to omit --sub-file completely (except when the naming scheme used for the subtitle files is overly disorganized):

```
 --sub-auto=<no|exact|fuzzy|all>, --no-sub-auto                                                                                                                                         
              Load additional subtitle files matching the video filename. The parameter specifies how external subtitle files are matched. exact is enabled by default.                       

              no     Don't automatically load external subtitle files.

              exact  Load the media filename with subtitle file extension (default).

              fuzzy  Load all subs containing media filename.

              all    Load all subs in the current and --sub-file-paths directories.
```

e.g.
```
sub-auto=fuzzy
```
I hope this works for you.

**Comment by luigino** [comment]
Well, I'm already using that option. My (small) complaint is that when I hit tab after "mpv foo", it tries to complete the first argument with the sub file as well and not just with the video one.

Not sure if that fixable.

**Comment by 2ion** [comment]
It is, with the same problem as explained above: which media file extensions to consider. The safest bet would be to look at the mime type of each file, but that's not an option for huge media libraries.

I'll add a runtime tunable variable by way of which you can enable a filter and customize the the media extension list (default off). Leaving this issue open till then.

```
MPV_BASH_COMPLETION_FILTER_MEDIAFILES=off # or on
MPV_BASH_COMPLETION_MEDIA_GLOB="<glob>"
```

**Comment by 2ion** [comment]
@luigino , your requested feature has been implemented in the new release 3.3.17 (https://github.com/2ion/mpv-bash-completion/releases/tag/3.3.17). If the OpenSuSE package maintainer pulls and packages that release for your distro is up to him though. If in doubt, you can get it manually (see README).
